cmd/main.go

```diff
@@ -325,8 +326,16 @@ func main() {
 	failOnError(err, "failed to create ignition builder")
 	installConfigBuilder := installcfg.NewInstallConfigBuilder(log.WithField("pkg", "installcfg"), mirrorRegistriesBuilder, providerRegistry)
 
+	compositeXattrClient, err := s3wrapper.NewCompositeXattrClient(
+		log,
+		s3wrapper.NewOSxAttrClient(log, filepath.Join(Options.WorkDir, Options.S3Config.S3Bucket)),
+		s3wrapper.NewFilesystemBasedXattrClient(log, filepath.Join(Options.WorkDir, Options.S3Config.S3Bucket)),
```

Comment:
You may want to avoid repeating the construction of the root directory in these ^ tool lines. Maybe do it before calling the `New...` function?